src/metabase/api/collection.clj

```diff
@@ -1063,8 +1063,19 @@
   "Implementation for the `dashboard-question-candidates` endpoints."
   [collection-id :- [:maybe ms/PositiveInt]]
   (api/check-403 api/*is-superuser?*)
-  (let [cards-in-collection (t2/hydrate (t2/select :model/Card {:where [:= :collection_id collection-id]}) :in_dashboards)]
-    (filter card/sole-dashboard-id cards-in-collection)))
+  (let [all-cards-in-collection (t2/hydrate (t2/select :model/Card {:where [:= :collection_id collection-id]}) :in_dashboards)]
+    (filter
+     (fn [card]
+       (and
+        ;; we're a good candidate if:
+        ;; - we're only in one dashboard
+        (card/sole-dashboard-id card)
+        ;; - that one dashboard is in the same collection
+        (= (:collection_id card)
+           (-> card :in_dashboards first :collection_id))
+        ;; - ... and we're not already a DQ in that dashboard
+        (nil? (:dashboard_id card))))
```

Comment:
small nit, but is this clause that dashboard_id is nil worth putting into the query?